src/common/ceph_argparse.cc

```diff
@@ -377,19 +377,19 @@ static int va_ceph_argparse_witharg(std::vector<const char*> &args,
 	std::ostream &oss, va_list ap)
 {
   const char *first = *i;
-  char tmp[strlen(first)+1];
-  dashes_to_underscores(first, tmp);
-  first = tmp;
+  std::vector<char> tmp(strlen(first) + 1);
+  dashes_to_underscores(first, tmp.data());
+  first = tmp.data();
 
   // does this argument match any of the possibilities?
   while (1) {
     const char *a = va_arg(ap, char*);
     if (a == NULL)
       return 0;
     int strlen_a = strlen(a);
-    char a2[strlen_a+1];
-    dashes_to_underscores(a, a2);
-    if (strncmp(a2, first, strlen(a2)) == 0) {
+    std::vector<char> a2(strlen_a+1);
+    dashes_to_underscores(a, a2.data());
+    if (strncmp(a2.data(), first, strlen(a2.data())) == 0) {
```

Comment:
If moved to strings, consider:
https://www.boost.org/doc/libs/1_87_0/doc/html/doxygen/structboost_1_1algorithm_1_1is__iequal.html

Reply:
Sometime in the future when modernizing, I think. Getting rid of all the low level string manipulation would be good as a separate PR.